src/backend/src/services/work-packages.services.ts

```diff
@@ -279,6 +280,70 @@ export default class WorkPackagesService {
     return `${created.wbsElement.carNumber}.${created.wbsElement.projectNumber}.${created.wbsElement.workPackageNumber}`;
   }
 
+  /**
+   * Creates a Work_Package_Template in the database
+   *
+   * @param user the user creating the work package template
+   * @param templateName the template name
+   * @param templateNotes the template notes
+   * @param workPackageName the name of the work packge
+   * @param stage the stage
+   * @param duration the duration of the work package template in weeks
+   * @param expectedActivities the expected activities descriptions for this WPT
+   * @param deliverables the expected deliverables descriptions for this WPT
+   * @param blockedBy the WBS elements that need to be completed before this WPT
+   * @returns the WBS number of the successfully created work package template
+   * @throws if the work package template could not be created
+   */
+  static async createWorkPackageTemplate(
+    user: User,
+    templateName: string,
+    templateNotes: string,
+    workPackageName: string | null,
+    stage: WorkPackageStage | null,
+    duration: number,
+    expectedActivities: string[],
+    deliverables: string[],
+    blockedBy: BlockedByInfo[]
+  ): Promise<WorkPackageTemplate> {
+    if (!isAdmin(user.role)) throw new AccessDeniedAdminOnlyException('create work package templates');
+
+    // get the corresponding IDs of all work package templates in BlockedBy,
+    // and throw an errror if the template doesn't exist
+    await Promise.all(
+      blockedBy.map(async (elem) => {
+        const template = await prisma.work_Package_Template.findFirst({
```

Comment:
this isnt a good idea, we should be identifying other templates by Id or something